### PR TITLE
[CI] Remove Azure workflows

### PR DESCRIPTION
**Pull request by akashchi**
Ticket: 142335.

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by akashchi** [comment]
I'll add the `overall_status` job as it is in the `openvino` repo so that if any of the jobs is skipped the required state is not an issue.

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -2,16 +2,10 @@ name: Linux (Ubuntu 20.04, Python 3.11)
 on:
   workflow_dispatch:
   pull_request:
-    paths-ignore:
-        - 'modules/nvidia_plugin'
-        - 'modules/openvino_code'
```
Comment (akashchi) [inline .github/workflows/linux.yml:7]: We do not need it as the NVIDIA plugin job was integrated into this workflow.

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -331,3 +325,17 @@ jobs:
 
       - name: Show ccache stats
         run: ccache --show-stats
+
+  Overall_Status:
+    name: ci/gha_overall_status_linux
```
Comment (akashchi) [inline .github/workflows/linux.yml:330]: It is not required as of now (as I've deleted the conditional start in the beginning of the file), but could be beneficial in the future.

**Comment by akashchi** [comment]
Should be merged after #919

**Comment by ilya-lavrenov** [comment]
We also need to update required checks for future release branches

**Pull request merged**